#TSS Annotation


/proj/Schmidbaur/Euprymna/liftover_la46/new_all_chroms_uniq.gff


1. Annotation of peaks to genes 
2. PASA to get TSS from transcript data generated previously 
* fasta files of transcript
* mapping of transcript to genome 
* prediction from transcript where intron exons are and which transcription direction 3'UTR and 5'UTR 
* in genome now only annotation for exons not TSS

[R Packages](https://rockefelleruniversity.github.io/RU_ATAC_Workshop.html)  

* 40 kb before exon should be promoter by programme default 



###1. PASA TSS annotation 

* we will use PASA for TSS annotation, previous transcriptome generated by madi? 
* uisng the information of the transcription prediction of TSS
* prediction is based on TSS being at the first exon 
* last exon for prediction of end site 
* transcripts of Euprymna (genome project) are in 
``/proj/Simakov/EUPRYMNA/Eup.final.fna``
* copy to working folder 
``` cp /proj/Simakov/EUPRYMNA/Eup.final.fna .```

* programme: ``` /scratch/Simakov/INSTALL/PASApipeline/```

* Config files: 
``` /scratch/Simakov/HYDRA/AEP/10X_Assemblies_fasta/v2.0.0_101X_fasta_Christy/pasa.config```

* to copy to working folder: 
```cp /scratch/Simakov/HYDRA/AEP/10X_Assemblies_fasta/v2.0.0_101X_fasta_Christy/pasa.config PASAconfig.txt ```

```
#!/bin/bash
#
#SBATCH --output=/scratch/hoang/PASA/PASA.o%A.%a
#SBATCH --error=/scratch/hoang/PASA/PASA.e%A.%a
#SBATCH --cpus-per-task=8
#SBATCH --mem=25000
#SBATCH --partition=basic
#SBATCH --nice=0
#SBATCH --job-name=PASA
#SBATCH --ntasks=1

module load gmap 
module load samtools 

perl /scratch/Simakov/INSTALL/PASApipeline/Launch_PASA_pipeline.pl -C -c PASAconfig.txt -R -g Eup_chr48.fasta -t Eup.final.fna --ALIGNERS gmap --CPU 8
```

* perl to tell the programme which language 
* C for flag/command create database
* c to specify the alignment assembly configuration file 
* R to run alignment 
* g to specify genome sequence fasta file 
	* the fasta file was copied to same folder 
	* using the path did not work 	
* t to specify transcript 
* aligners in gmap option
	* GMAP and/or BLAT is used to map and align the transcripts to the genome	
* CPU used here 8 


Config file: 

* some programmes need a file in which the configuration are prefined 

~~~
## templated variables to be replaced exist as <__var_name__>

# database settings
DATABASE=/scratch/hoang/PASA/PASAdb.sqlite

#######################################################
# Parameters to specify to specific scripts in pipeline
# create a key = "script_name" + ":" + "parameter" 
# assign a value as done above.

#script validate_alignments_in_db.dbi
validate_alignments_in_db.dbi:--MIN_PERCENT_ALIGNED=80
validate_alignments_in_db.dbi:--MIN_AVG_PER_ID=90

#script subcluster_builder.dbi
subcluster_builder.dbi:-m=50
~~~

* error message was shown, transcript big 
* Oleg changed somethings on the code

Next: run as slurm 

```
perl /scratch/Simakov/INSTALL/PASApipeline/scripts/pasa_asmbls_to_training_set.dbi --pasa_transcripts_fasta PASAdb.sqlite.assembli
es.fasta --pasa_transcripts_gff3 PASAdb.sqlite.pasa_assemblies.gff3
```

* PASA generated a gff3 file in which we still had to change the 
